[KF-6704;KF-5958;KF-6782] Bundle version checks + Upgrade to Python 3.10+ using Poetry

deusebio opened the pull request:
Backporting:
* https://github.com/canonical/charmed-kubeflow-uats/pull/137
* https://github.com/canonical/charmed-kubeflow-uats/pull/155

This has been tested on [AWS EKS](https://github.com/canonical/bundle-kubeflow/actions/runs/12932149755)

misohu commented on driver/test_kubeflow_workloads.py:
```diff
@@ -55,22 +56,56 @@
 
 KFP_PODDEFAULT_NAME = "access-ml-pipeline"
 
+CHARMS = {
```
Same comment as here https://github.com/canonical/charmed-kubeflow-uats/pull/155#discussion_r1935178781

misohu reviewed (commented):
Before merging we need to merge this https://github.com/canonical/charmed-kubeflow-uats/pull/155

Also I think we should split this into two separate bacport PRs. We have two separate PRs to main so we should also do it here. Separate backport prs are useful in case we need some rollbacks in the future.

deusebio commented:
> Also I think we should split this into two separate bacport PRs.

I have rebased and squashed commits in order to have two well-perimetered commits for each backport, that we can rebase and merge. I would not open two separate PRs as this also would required the burden for EKS testing and rebase (and wait for one PR to land before merging/testing the next one), with no sensible benefits

[EKS tests currently running](https://github.com/canonical/bundle-kubeflow/actions/runs/12932149755)

deusebio commented:
yes, I believe it is a good practice. We do the same, but sometimes, in peculiar cases, we use also rebase & merge, like in this case or when dealing with feature branches, where we have well-perimetered tickets in the user-story that we want to keep their granularity in the main branch as well

orfeas-k commented:
@deusebio In the team, we are trying to follow https://www.conventionalcommits.org/en/v1.0.0/ convention for commits going the branch history. I think it makes sense to address this in the PR. If that's too much of a burden, let's keep it as a guideline for future work.

deusebio commented:
@orfeas-k Not a burden but given that the two commits have landed with those names already in the main branch (here we are just backporting), doing this here only would seem a bit inconsistent. 

Do we have a convention/suggestion for the merge commit?
